core/modules/widgets/navigator.js

```diff
@@ -146,22 +158,35 @@ NavigatorWidget.prototype.handleNavigateEvent = function(event) {
 NavigatorWidget.prototype.handleCloseTiddlerEvent = function(event) {
 	var title = event.param || event.tiddlerTitle,
 		storyList = this.getStoryList();
+	var adjacentTiddler = this.historyTitle ? this.findAdjacentStoryTiddler(title,storyList) : null;
 	// Look for tiddlers with this title to close
 	this.removeTitleFromStory(storyList,title);
 	this.saveStoryList(storyList);
+	if(adjacentTiddler) {
+		this.wiki.setText(this.historyTitle,"previous-focus-tiddler",undefined,this.wiki.getTiddler(this.historyTitle).getFieldString("focus-tiddler"));
+		this.wiki.setText(this.historyTitle,"focus-tiddler",undefined,adjacentTiddler);
+	}
```

Comment:
I think this should be handled at a lower level. follow the `.addHistory()` function, down too the root. It should be handled there

Reply:
the underlying problem is that addToHistory is not called when closing, deleting or cancelling a tiddler (and the tiddler is removed)